pkg/prometheus/server/operator.go

```diff
@@ -1226,3 +1230,45 @@ func makeSelectorLabels(name string) map[string]string {
 		"prometheus":                    name,
 	}
 }
+
+func validateAlertmanagerEndpoints(am monitoringv1.AlertmanagerEndpoints) error {
+	var nonNilFields []string
+
+	//nolint:staticcheck // Ignore SA1019 this field is marked as deprecated.
+	if am.BearerTokenFile != "" {
+		nonNilFields = append(nonNilFields, fmt.Sprintf("%q", "bearerTokenFile"))
+	}
+
+	for k, v := range map[string]interface{}{
+		"basicAuth":     am.BasicAuth,
+		"authorization": am.Authorization,
+		"sigv4":         am.Sigv4,
+	} {
+		if reflect.ValueOf(v).IsNil() {
+			continue
+		}
+		nonNilFields = append(nonNilFields, fmt.Sprintf("%q", k))
+	}
+
+	if len(nonNilFields) > 1 {
+		return fmt.Errorf("%s can't be set at the same time, at most one of them must be defined", strings.Join(nonNilFields, " and "))
+	}
+
```

Comment:
I am unable to call `validateRelabelConfigs()` inside the relocated `validateAlertmanagerEndpoints()`, what should we do? now that we have both `alertRelabelConfigs` and `relabelConfigs`?